packages/native/src/useLinking.tsx

```diff
@@ -40,6 +40,21 @@ export default function useLinking(
   ref: React.RefObject<NavigationContainerRef>,
   { prefixes, config, getStateFromPath = getStateFromPathDefault }: Options
 ) {
+  React.useEffect(() => {
+    const listenerRef = ref?.current?.__isListenerSetRef;
+    if (listenerRef) {
+      if (listenerRef.current === false) {
+        listenerRef.current = true;
+      } else {
+        throw new Error(`Multiple instances of 'useLinking' are used`);
```

Comment:
```suggestion
        throw new Error("Looks like you are using 'useLinking' in multiple components. This is likely an error since deep links need to be handled only once.");
```